06week/higherOrder.js

```diff
@@ -2,18 +2,45 @@
 
 const assert = require('assert');
 
-function forEach(arr, callback) {
-  // Your code here
-}
-
-function map(arr, callback) {
-  // Your code here
-}
-
-function filter(arr, callback) {
-  // Your code here
-}
 
+// function forEach(arr, callback){
+const forEach = (inputArray, inputFunction) => {
+  const anotherArr = [];
+  for (let i = 0; i < inputArray.length; i++){
+    anotherArr.push(inputFunction(inputArray[i]));
+    console.log(inputArray.length)
+  }
+    // return anotherArr;
+};
+const fullArray = [1, 2, 3];
+const inputFunction = i => fullArray.length;
+forEach(fullArray, inputFunction);
+// function map(arr, callback) {
+//
+// }
+const map = (arg1, arg2) => {
+  let nums = [];
+  arg1.forEach(index => nums.push(arg2(index)));
```

Comment:
This map function works perfectly, but some of the things inside aren't named well. `index` isn't the index, it's the item, and the return array won't always be full of numbers, so `nums` doesn't make sense. But it works!